js/index.js

```diff
@@ -31,4 +32,156 @@ navItems.forEach(
+      img: './ressources/Portfolioshow.png',
+    },
+  ];
+  
+  /**
+   * code for popup window
+   */
+  
+   modalData.forEach((project, index) =>{
+    let lang ="";
+    let identity ="";
+    if(project.main){
+    
+      project.languages.forEach((element) => {
+        lang += `<li class="language-list-item">${element}</li>`;
+      });
+      document.querySelector('#main-grid').innerHTML = `
+          <div class="works-section-project">
+            <img src="${project.img}" alt="${project.name}">
+          </div>
+          <div class="works-section-project-description">
+            <h2 class="works-section-project-header">
+              ${project.name}
+            </h2>
+            <p class="work-project-para">
+              ${project.description} ${project.description} 
+            </p>
+            <ul class="language-list-body">
+              ${lang}
+            </ul>
+            <button data-index="${index}" type="button" class="project-button button display-modal popup-btn">See Project</button>
+          </div>`;
+    }else{
+      project.languages.forEach((element) => {
+        lang += `<li class="language-list-item works-grid-one-project-language-list">${element}</li>`;
+      });
+      document.querySelector('#secondary-grid').innerHTML += `
+        <div class="works-grid-one-project-container first-grid style="background-image: url('${project.img}')">
+          <div class="grid-project1 works-grid-project">
+            <h3 class="works-grid-one-project-header">${project.name}</h3>
+            <p class="works-grid-one-project-description">
+              ${project.description}
+            </p>
+  
+            <ul class="language-list-body">
+            ${lang}
+            </ul>
+          </div>
+          <button type="button" data-index="${index}" class="grid-project-button display-modal button">See Project</button>
+        </div>`;
+    }
+    
+  });
+  
+  const openPopup = (index) => {
+    let lang = '';
+    modalData[index].languages.forEach((element) => {
+      lang += `<li class="language-list-item">${element}</li>`;
+    });
+    document.querySelector('.modal-container').style.display = 'block';
+    document.querySelector('#popuptitle').textContent = modalData[index].name;
+    document.querySelector('#modallanguages').innerHTML = lang;
+    document.querySelector('#modal-description').textContent = modalData[index].description;
+    document
+      .querySelector('#modal-img')
+      .setAttribute('src', modalData[index].img);
+      document
+      .querySelector('#modal-img')
+      .setAttribute('alt', modalData[index].name);
+  };
+  
+  const modalDisplayBtn = document.querySelectorAll('.display-modal');
+  
+  modalDisplayBtn.forEach((element) => {
+    element.addEventListener('click', () => {
+      openPopup(element.getAttribute('data-index'));
+    });
+  });
+  
```

Comment:
- [ ] [OPTIONAL] I kindly suggest that after displaying a modal, you add an overflow of hidden to the web page, so that the user does not accidentally scroll away from where he was before